javascript/ql/src/Expressions/HeterogeneousComparison.ql

```diff
@@ -170,13 +170,38 @@ string getTypeDescription(string message1, string message2, int complexity1, int
     result = message1
 }
 
+/**
+ * Holds if `e` directly uses a parameter's initial value as passed in from the caller.
+ */
+predicate isInitialParameterUse(Expr e) {
+  // unlike `SimpleParameter.getAnInitialUse` this will not include uses we have refinement information for
+  exists (SimpleParameter p, SsaExplicitDefinition ssa |
+    ssa.getDef() = p and
+    ssa.getVariable().getAUse() = e and
+    not p.isRestParameter()
+  )
+}
+
+/**
+ * Holds if `e` is an expression that should not be considered in a heterogeneous comparison.
+ *
+ * We currently whitelist these kinds of expressions:
+ *
+ *   - parameters, as passed in from the caller
```

Comment:
This is too terse. Explain what "as passed in from the caller" means, and explain why we whitelist this. Also, let's not have one-element lists.